### PR TITLE
Support overriding of scm-source.json via X-SCM-Source header, but on…

### PR DESCRIPTION
**Pull request by dryewo**
…ly when application.write_all_trusted scope is set.

This is used by CI to avoid requiring the users to include `/scm-source.json` file (and to avoid relying on it).

### CODE REVIEW

File: resources/db/pierone.sql [hunk 1]
```diff
@@ -74,6 +74,21 @@ SELECT ssd_url AS url,
 ORDER BY created DESC
  LIMIT 1;
 
+-- name: create-or-update-scm-source-data!
+WITH scm_source_data_update AS (
```
Comment (hjacobs) [inline resources/db/pierone.sql:78]: we could use the new PostgreSQL upsert

**Comment by juhmelo** [comment]
:+1:

File: resources/db/pierone.sql (hunk 1 above)
Comment (dryewo) [reply resources/db/pierone.sql:78]: Good point, but current RDS instance runs 9.4. I'd rather not upgrade it right now.

**Comment by dryewo** [comment]
👍

**Pull request merged**